src/common/wincmn.cpp

```diff
@@ -713,11 +703,7 @@ wxSize wxWindowBase::DoGetBestSize() const
         best = wxSize(maxX, maxY);
     }
 #endif // wxUSE_CONSTRAINTS
-    else if ( !GetChildren().empty()
-#if defined( __WXMAC__ ) && !defined(__WXUNIVERSAL__)
-              && wxHasRealChildren(this)
-#endif
-              )
+    else if ( wxHasRealChildren(this) )
```

Comment:
Looking at this code once again, I think we don't actually need this at all as we redo the same tests in the loop inside anyhow.

So we could just remember if we have any children there and fallback to the code in the `else` branch if not. IMO this would make things more clear, as right now we have similar but not quite identical checks in 2 places.